src/libYARP_companion/src/yarp/companion/impl/Companion.cmdTrafficGen.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2006-2020 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#include <yarp/companion/impl/Companion.h>
+
+#include <yarp/os/Bottle.h>
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/Network.h>
+#include <yarp/os/Property.h>
+#include <yarp/os/PeriodicThread.h>
+#include <limits>
+
+using yarp::companion::impl::Companion;
+using yarp::os::Bottle;
+using yarp::os::BufferedPort;
+using yarp::os::NetworkBase;
+using yarp::os::Property;
+using namespace std;
+
+class datasender : public yarp::os::PeriodicThread
+{
+private:
+    double m_data_size_MB;
+    char* data_buff =nullptr;
+    std::string m_portname;
+    yarp::os::Port outport;
+    yarp::os::Bottle pp;
+
+public:
+    datasender(double period, string portname, double data_size_MB) : PeriodicThread (period)
+    {
+        m_portname=portname;
+        m_data_size_MB = data_size_MB;
```

Comment:
This should be done using the member initialization instead of assignment in the constructor